airflow/contrib/executors/kubernetes_executor.py

```diff
@@ -476,6 +480,27 @@ def _make_safe_pod_id(safe_dag_id, safe_task_id, safe_uuid):
 
         return safe_pod_id
 
+    @staticmethod
+    def _make_safe_label_value(string):
+        """
+        Valid label values must be 63 characters or less and must be empty or begin and
+        end with an alphanumeric character ([a-z0-9A-Z]) with dashes (-), underscores (_),
+        dots (.), and alphanumerics between.
+
+        If the label value is then greater than 63 chars once made safe, or differs in any
+        way from the original value sent to this function, then we need to truncate to
+        53chars, and append it with a unique hash.
+        """
+        MAX_LABEL_LEN = 63
+
+        safe_label = re.sub(r'^[^a-z0-9A-Z]*|[^a-zA-Z0-9_\-\.]|[^a-z0-9A-Z]*$', '', string)
```

Comment:
so if I'm understanding this correctly this will prevent any collisions. So if I have "my*favorite*dag" and "my^favorite^dag", then one will be "myfavoritdag<hash1>" and the second will be  "myfavoritedag<hash2>". Is that correct?

Reply:
Yes that is correct.  As the regex and formatting syntax on the kubernetes docs is quite precise, if we were to simply hash only if the modified labels were above 63 chars could lead to collisions if there are unsupported chars being the only difference between the dags (as you've mentioned above).  This also covers accentuated characters if they are used in dag names too.